client/components/open/forms/OpenCompleteForm.vue

```diff
@@ -288,14 +300,16 @@ export default {
             id: this.form.id,
             redirect_target_url: (this.form.is_pro && data.redirect && data.redirect_url) ? data.redirect_url : null
           },
-          submission_data: form.data()
+          submission_data: form.data(),
+          completion_time: completionTime
         })
 
         if (this.isIframe) {
           window.parent.postMessage(payload, '*')
         }
         window.postMessage(payload, '*')
         this.pendingSubmission.remove()
+        this.pendingSubmission.removeTimer()
```

Comment:
Great but if validation failed (or other) then probably we should continue the timer